src/dune_rules/package.ml

```diff
@@ -80,15 +80,13 @@ module Dependency = struct
     let map =
       [ ("=", Eq); (">=", Gte); ("<=", Lte); (">", Gt); ("<", Lt); ("<>", Neq) ]
 
-    let to_dyn =
-      let open Dyn in
-      function
-      | Eq -> string "Eq"
-      | Gt -> string "Gt"
-      | Gte -> string "Gte"
-      | Lte -> string "Lte"
-      | Lt -> string "Lt"
-      | Neq -> string "Neq"
+    let to_dyn = function
+      | Eq -> Dyn.variant "Eq" []
+      | Gte -> Dyn.variant "Gte" []
+      | Lte -> Dyn.variant "Lte" []
+      | Gt -> Dyn.variant "Gt" []
+      | Lt -> Dyn.variant "Lt" []
+      | Neq -> Dyn.variant "Neq" []
```

Comment:
You can also just look it up from `map`. Save some code duplication this way.

Reply:
However we won't get the constructor names this way and only the symbols.;